### ISSUE TITLE
non-repeatable test failure in orphan_s3_originals_spec

### ISSUE THREAD

**Issue by jrochkind** [sciencehistory/scihist_digicoll#1686]
This failure can happen, and then not happen again re-running CI with no code changes. Some kind of race condition going on. We need to fix it, or disable the test. 

Pretty mysterious to me what's going on here, not really sure. 

```
  1) OrphanS3Originals two legit assets, one video and one image, plus two orphans deletes the orphans
     Failure/Error: iter.shrine_storage.delete(shrine_path)

     NoMethodError:
       undefined method `gsub' for nil:NilClass
     # ./app/services/orphan_s3_originals.rb:97:in `block (2 levels) in delete_orphans'
     # ./app/services/s3_path_iterator.rb:73:in `block in each_s3_path'
     # ./app/services/s3_path_iterator.rb:115:in `block (2 levels) in _each_s3_path'
     # ./app/services/s3_path_iterator.rb:114:in `each'
     # ./app/services/s3_path_iterator.rb:114:in `block in _each_s3_path'
     # ./app/services/s3_path_iterator.rb:110:in `_each_s3_path'
     # ./app/services/s3_path_iterator.rb:72:in `each_s3_path'
     # ./app/services/orphan_s3_originals.rb:94:in `block in delete_orphans'
     # ./app/services/orphan_s3_originals.rb:91:in `each'
     # ./app/services/orphan_s3_originals.rb:91:in `delete_orphans'
     # ./spec/services/orphan_s3_originals_spec.rb:70:in `block (3 levels) in <main>'

Failed examples:

rspec ./spec/services/orphan_s3_originals_spec.rb:67 # OrphanS3Originals two legit assets, one video and one image, plus two orphans deletes the orphans
```
https://github.com/sciencehistory/scihist_digicoll/blob/5a3f9121a4a20db8aba8ce341a112f5e6a6a8122/spec/services/orphan_s3_originals_spec.rb#L67

https://github.com/sciencehistory/scihist_digicoll/blob/5a3f9121a4a20db8aba8ce341a112f5e6a6a8122/app/services/orphan_s3_originals.rb#L97

**Comment by eddierubeiz** [comment]
The problem has to do with the use of `inline_promoted_file` assets. No fix yet, but the PR above provides a simple demonstration of the problem and a workaround.
I also believe I can rewrite this test to not use `inline_promoted_file assets`, regardless of whether and how we fix the problem.

**Comment by eddierubeiz** [comment]
This was fixed by #1714.
